File: CS 3060 Projects/Homework1b/Task1/Task1.rb
# Open the file in write mode to save user input
File.open("myFile.txt", "w") do |file|
  line_number = 1

  # Ask the user for 4 lines of input
  4.times do
    puts "Enter line ##{line_number}:"
    line = gets.chomp

    # Write the input to the file with line number
    file.puts "L#{line_number}: #{line}"

    line_number += 1
  end
end

# Open the file in read mode to process and count words
File.open("myFile.txt", "r") do |file|
  line_number = 1

  # Process each line in the file
  file.each_line do |line|
    words = line.split
    word_count = words.size - 1

    # Output the word count for each line
    puts "L#{line_number}: #{word_count}"

    line_number += 1
  end
end

File: CS 3060 Projects/Homework3/Task1/task1.scala
import scala.io.StdIn
import java.io.PrintWriter

object Task1 {
  def main(args: Array[String]): Unit = {
    // Step 1: Ask user for input and save it to a file
    println("Please type 4 lines:")
    val lines = (1 to 4).map(_ => StdIn.readLine())

    val writer = new PrintWriter("fileTask1.txt")
    lines.foreach(writer.println)
    writer.close()

    // Step 2: Read the file and count words and vowels
    val source = scala.io.Source.fromFile("fileTask1.txt")
    val content = source.mkString
    source.close()

    val wordCount = content.split("\\s+").length
    val vowelCount = content.toLowerCase.count(c => "aeiou".contains(c))

    // Step 3: Report the results
    println(s"Number of words: $wordCount")
    println(s"Total number of vowels: $vowelCount")
  }
}
